lib/surface/compiler/eex_engine.ex

```diff
@@ -723,6 +671,10 @@ defmodule Surface.Compiler.EExEngine do
     ]
   end
 
+  # TODO: [Type] We should probably convert an attribute value containing interpolation
```

Comment:
I agree with this - normalizing the attribute values to either a single `AST.Text` or a single `AST.AttributeExpr` would simplify handling this throughout the rest of the code. 

It may make sense to handle lists of values in the `TypeHandler`, instead of directly in the compiler - the handling for them is somewhat type specific (i.e. a string makes sense to just join, but what does it mean for a css_class property to be specified as `"str_1 {{ @str_2 }}"`, and some (keyword, map, list) are probably just plain errors). I could see adding another method (maybe `list_expr_to_quoted!`?) that delegated handling of individual `AttributeExpr` to `expr_to_quoted!`, then called a similar method on the type handler to convert the list of `Text` and `AttributeExpr` into a single `AttributeExpr`.

This could potentially handle the empty list case as well (where the user specifies an empty string as the value - i.e. `class=""`), and could allow us to convert the list of expressions and/or text nodes into a single expression or text node.

Reply:
We'd have to add a way for the type handlers to process literal values (similar to your suggested `literal_to_html`, but used at the same stage as `expr_to_quoted!` instead of `attr_to_html`

Reply:
> what does it mean for a css_class property to be specified as `"str_1 {{ @str_2 }}"`, and some (keyword, map, list) are probably just plain errors)

I think we shouldn't accept that at all! An attribute value wrapped in `""` is a string and as such, it should behave just like any other string in Elixir. The interpolation should work the same way as `#{}` works, i.e. it should accept only:

  * strings
  * integers representing Unicode code points
  * a list containing one of these three elements
  *  values of types that implement the `String.Chars` protocol

We could even drop support for `{{...}}` inside attributes values to force the user to use `#{}` for interpolation. That would make it work just like React, which only accept interpolation on attributes using JS expressions like: 

```
title={"Hello " + name}
````
 or 
```js
title={`Hello ${name}`}
```

I find the `{{...}}` nicer than `#{}` in the context of a template because it provides a standard way for interpolation. However, I'd have no problem removing it if it would make Surface more consistent :)

> We'd have to add a way for the type handlers to process literal values (similar to your suggested `literal_to_html`, but used at the same stage as `expr_to_quoted!` instead of `attr_to_html`.

I agree. That would also allow us to easily provide static type-based validation for literals, which is pretty cool.